### ISSUE TITLE
load-data-readers! in pods

### ISSUE THREAD

**Issue by martinklepsch** [boot-clj/boot#535]
An https://github.com/levand/immuconf asked in Slack how he can access config variables during compilation. A macro would usually do the trick but in this case the pod compiling the ClojureScript sources and running the macro does not have the reader function for the immuconf specific tags.

If I understand the issue correctly I see the following options

- We set `*data-readers*` in pods after initialization
- We move `boot.core/load-data-readers!` into a namespace that is available in pods and advise users to call it if they need support for dependency-imported reader tags.

**I didn't reproduce this issue so maybe it's absolute bogus, just from the initial report it seemed plausible though**

**Comment by arichiardi** [comment]
About this, I just tried:

```
(pod/with-eval-in pod
      (#'clojure.core/load-data-readers)
      (set! *data-readers* (.getRawRoot #'*data-readers*)))
```

And:

```
java.lang.IllegalStateException: Can't change/establish root binding of: *data-readers* with set
```

**Comment by martinklepsch** [comment]
Maybe we document the `binding` solution presented above in the [Troubleshooting](https://github.com/boot-clj/boot/wiki/Boot-Troubleshooting) page and close this issue?

**Comment by martinklepsch** [comment]
@arichiardi Are you suggesting the following?
```
(pod/with-eval-in pod
  (#'clojure.core/load-data-readers)
  (binding [*data-readers* (.getRawRoot #'*data-readers*)]
    (do (some-stuff)))
```
If that works we should probably document it in the [Troubleshooting page](https://github.com/boot-clj/boot/wiki/Boot-Troubleshooting) and close this issue.

I'll try to remember who was that immuconf user. No idea why I didn't write it down in the first place.

**Comment by arichiardi** [comment]
Yes sorry I was a bit in a rush, I tried the above indeed and it was not complaining, I did not test the particular issue though.
